@alyssais it seems `cryptsetup` fails to build in its tests:

```
FAILED backtrace:
276 ./compat-test
FAIL: compat-test
[…]
=======================
1 of 6 tests failed
(23 tests were not run)
=======================
make[3]: *** [Makefile:802: check-TESTS] Error 1
make[3]: Leaving directory '/build/cryptsetup-2.4.3/tests'
make[2]: *** [Makefile:928: check-am] Error 2
make[2]: Leaving directory '/build/cryptsetup-2.4.3/tests'
make[1]: *** [Makefile:930: check] Error 2
make[1]: Leaving directory '/build/cryptsetup-2.4.3/tests'
make: *** [Makefile:2604: check-recursive] Error 1
error: builder for '/nix/store/a7ycpsqi9c9lgflhf6rc3yib5drqa74c-cryptsetup-2.4.3.drv' failed with exit code 2;
       last 10 log lines:
       > 1 of 6 tests failed
       > (23 tests were not run)
       > =======================
       > make[3]: *** [Makefile:802: check-TESTS] Error 1
       > make[3]: Leaving directory '/build/cryptsetup-2.4.3/tests'
       > make[2]: *** [Makefile:928: check-am] Error 2
       > make[2]: Leaving directory '/build/cryptsetup-2.4.3/tests'
       > make[1]: *** [Makefile:930: check] Error 2
       > make[1]: Leaving directory '/build/cryptsetup-2.4.3/tests'
       > make: *** [Makefile:2604: check-recursive] Error 1
       For full logs, run 'nix log /nix/store/a7ycpsqi9c9lgflhf6rc3yib5drqa74c-cryptsetup-2.4.3.drv'.
error: 1 dependencies of derivation '/nix/store/n0spz83536vqb036d2x2fn7pmmkwc0qz-systemd-250.4.drv' failed to build
```

--- a/pkgs/os-specific/linux/systemd/default.nix
+++ b/pkgs/os-specific/linux/systemd/default.nix
@@ -209,6 +209,8 @@ stdenv.mkDerivation {
       (musl-patches + "/0025-Handle-__cpu_mask-usage.patch")
       (musl-patches + "/0026-Handle-missing-gshadow.patch")
       (musl-patches + "/0028-missing_syscall.h-Define-MIPS-ABI-defines-for-musl.patch")
+      (musl-patches + "/0001-pass-correct-parameters-to-getdents64.patch")
+      (musl-patches + "/0002-Add-sys-stat.h-for-S_IFDIR.patch")
       (musl-patches + "/0001-Adjust-for-musl-headers.patch")
     ]
   );

